site/src/sphinx/server-decorator.rst

```diff
@@ -42,22 +43,22 @@ a decorating service. It enables you to write a decorating service with a single
         return delegate.serve(ctx, req);
     });
 
-Extending ``SimpleDecoratingService``
--------------------------------------
+Extending ``SimpleDecoratingHttpService`` and ``SimpleDecoratingRpcService``
+----------------------------------------------------------------------------
 
 If your decorator is expected to be reusable, it is recommended to define a new top-level class that extends
-:api:`SimpleDecoratingService` :
+:api:`SimpleDecoratingHttpService` or :api:`SimpleDecoratingRpcService` depends on the service type:
```

Comment:
depends .. -> `depending on if you are decorating an :api:`HttpService` or an :api:`RpcService`.`